app/src/main/java/org/sil/storyproducer/controller/storylist/StoryPageFragment.kt

```diff
@@ -0,0 +1,151 @@
+package org.sil.storyproducer.controller.storylist
+
+import android.app.Activity
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import android.text.Html
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentTransaction
+import org.sil.storyproducer.R
+import org.sil.storyproducer.activities.BaseActivity
+import org.sil.storyproducer.controller.MainActivity
+import org.sil.storyproducer.model.Story
+import org.sil.storyproducer.service.SlideService
+
+
+class StoryPageFragment : Fragment() {
+
+    private lateinit var storyPageTab : StoryPageTab
+    private lateinit var listView: ListView
+    private lateinit var adapter: ListAdapter
+
+    companion object {
+        const val ARG_POSITION = "position"
+
+        fun getInstance(position: Int): StoryPageFragment {
+            val storyPageFragment = StoryPageFragment()
+            val bundle = Bundle()
+            bundle.putInt(ARG_POSITION, position)
+            storyPageFragment.arguments = bundle
+            return storyPageFragment
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        super.onCreate(savedInstanceState)
+
+        val position = requireArguments().getInt(ARG_POSITION)
+        storyPageTab = StoryPageTab.values()[position]
+
+        if (storyPageTab.getStoryList().isEmpty()) {
+            val view = inflater.inflate(R.layout.fragment_no_stories, container, false)
+
+            view!!.findViewById<TextView>(R.id.stories_not_found_text).text =
+                    if (Build.VERSION.SDK_INT >= 24){
+                        Html.fromHtml(getString(storyPageTab.emptyStoryStringId), 0)}
+                    else{
+                        Html.fromHtml(getString(storyPageTab.emptyStoryStringId))}
+
+            view.findViewById<Button>(R.id.update_workspace_button).setOnClickListener {
+                (activity as? BaseActivity)?.showSelectTemplatesFolderDialog()
```

Comment:
For an empty list - the update workspace button should only show in the All Stories tab, not in the other tabs. Does the layout here accommodate that?

Reply:
Ah I must have forgot that was part of your discussions with Claire. Yes this is possible, however it will take a few more tweaks because the current layout contains the button. fragment_no_stories.xml will need to be changed as well.

Reply:
On second thought, it would be very simple to hide the button. Here you go!
<img width="522" alt="Screen Shot 2021-04-13 at 12 25 20 AM" src="https://user-images.githubusercontent.com/15131163/114496920-f59a8b80-9bee-11eb-8df5-a20778fa5f8f.png">
<img width="522" alt="Screen Shot 2021-04-13 at 12 25 30 AM" src="https://user-images.githubusercontent.com/15131163/114496925-f6cbb880-9bee-11eb-8864-3143c24dc0f7.png">
<img width="522" alt="Screen Shot 2021-04-13 at 12 25 35 AM" src="https://user-images.githubusercontent.com/15131163/114496928-f7fce580-9bee-11eb-8fc6-c8ebd5e6d1b6.png">